app/javascript/profileDropdown/flagButton.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-alert */
+/**
+ * Adds a flag button visible only to trusted users on profile pages.
+ * @function initFlag
+ * @returns {(void|undefined)} This function has no useable return value.
+ */
+export function initFlag() {
+  const flagButton = document.getElementById(
+    'user-profile-dropdownmenu-flag-button',
+  );
+
+  if (!flagButton) {
+    // button not always present when this is called
+    return;
+  }
+
+  /* eslint-disable-next-line no-undef */
+  const user = userData();
+  if (!user) {
+    return;
+  }
+
+  const { profileUserId, profileUserName } = flagButton.dataset;
+  let flagStatus = flagButton.dataset.flagStatus === 'true';
```

Comment:
(not-blocking) - might just be me, but I think naming this something like `isUserFlagged` would make things clearer

Reply:
I have no strong preference so you saying you find it clearer is enough of an argument to rename it for me. Thanks for the suggestion 👍